quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// s2n-quic's default implementation of the datagram component
+
+use crate::{
+    datagram::unreliable_datagram::{ConnectionInfo, Endpoint, Packet, Receiver, Sender},
+    time::Timestamp,
+};
+use alloc::collections::VecDeque;
+use bytes::Bytes;
+use core::time::Duration;
+
+#[derive(Debug, Default)]
+pub struct Disabled;
+
+// Default capacity is zero right now since the feature is currently disabled.
+const DEFAULT_CAPACITY: usize = 0;
+
+impl Endpoint for Disabled {
+    type Sender = DefaultSender;
```

Comment:
Why does the disabled endpoint have an enabled sender? I think we need two endpoints: `Disabled` and `Default`